back/sssscs/src/main/java/com/ib/config/ServerConfig.java

```diff
@@ -57,9 +56,8 @@ public void addCorsMappings(CorsRegistry registry) {
 	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 		http.cors().and().csrf().disable();
 		http.headers().frameOptions().disable();
-		http.authorizeHttpRequests()
-			.requestMatchers("/api/user/session/**").permitAll()
-			.anyRequest().authenticated();
+		http.authorizeHttpRequests().requestMatchers("/api/user/session/**").permitAll()
+				.requestMatchers("/api/verification-code/**").permitAll().anyRequest().authenticated();
```

Comment:
add `requestMatchers("/*").permitAll()`, without it we can't access the h2 console

Reply:
Where inside the chain should I add it? I'm getting a 401 when trying to access `localhost:8080/h2-console`.

Reply:
Anywhere before `anyRequest` should be fine. Don't think it should mess with the other matchers.

Reply:
Okay, I think the problem was `requestMatchers("/*")` instead of `requestMatchers("/**")`.
[67725c9](https://github.com/magley/sssscs/pull/18/commits/67725c91798280d6424512cc48d518fdaf07b074)

Reply:
Nope, if I remember correctly what you just did is allow _all_ urls. This overrides any subsequent matchers, including the `anyRequest` which requries unspecified urls to be authenticated.

Reply:
I just tried several authorized endpoints in postman (`/api/cert/request/created`, `/api/cert/request`) with no authorization and it gave me a 401 Unauthorized response.

Reply:
I am confused, because from reading the code, it seems that everything should be permitted. I think `@PreAuthorize` overrides the chain which is why we still need authorization on all the endpoints (except `@PermitAll` and `/h2-console`), but this is a blacklist approach which is the opposite of what we want.

Reply:
Weird, I remember playing around with the asterisk or two asterisk dilemma last project and deciding that single asterisk is correct... I guess it's fine then.